packages/calypso-e2e/src/lib/blocks/image-block.ts

```diff
@@ -63,6 +72,40 @@ export class ImageBlock {
 		return await this.getImage();
 	}
 
+	/**
+	 * Given path to an image file, uploads the image to an Image block
+	 * using the media modal via `Select Images` > `Media Library`.
+	 *
+	 * @param {string} path Path to the image file.
+	 */
+	async uploadFromModal( path: string ): Promise< ElementHandle > {
+		const frame = ( await this.block.ownerFrame() ) as Frame;
+		const page = frame.page();
+
+		// The resulting React modal is essentially the MediaPage, just
+		// embedded within the modal, with additional buttons to
+		// confirm or cancel image insertion.
+		const mediaPage = new MediaPage( page );
+		await mediaPage.upload( path );
+		await mediaPage.clickModalButton( 'Confirm' );
+		await this.waitUntilUploaded();
+		return await this.getImage();
```

Comment:
This could/should probably be a different PR, but what do you think of moving these media blocks over to the block flow smoke test format soon? 

I was just thinking that it seems like a lot of work here to try to get the page, and moving over the `BlockFlow` interface might make that a little simpler! It also might be good to eventually break the dependence on passing around the element handle for the block.

Reply:
I intentionally didn't want to migrate everything to the block testing format to keep the scope of this PR tight. With that said, how would one implement all sorts of disparate actions with the block testing format?

In my mind, the block testing flow works because we're only performing one or two actions out of all possible actions on the block. Furthermore, the data that's input and the actions performed on the block themselves are on 'rails' so to speak.

In contrast, in these tests we're looking to test various aspects of the image upload process. 

I suppose the different actions can be handled in the block flow by placing multiple `if/else` branches?

My primary concern is that by using the block testing flow, we lose the flexibility to string together actions against the block to run many different scenarios.

Reply:
Filed a new task at https://github.com/Automattic/wp-calypso/issues/61140.